lib/improver/tests/nowcasting/optical_flow/test_AdvectField.py

```diff
@@ -246,16 +247,24 @@ def setUp(self):
         data = np.array([[2., 3., 4.],
                          [1., 2., 3.],
                          [0., 1., 2.],
-                         [0., 0., 1.]])
-        self.cube = iris.cube.Cube(
-            data, standard_name='rainfall_rate', units='mm h-1',
-            dim_coords_and_dims=[(self.plugin.y_coord, 0),
-                                 (self.plugin.x_coord, 1)])
+                         [0., 0., 1.]], dtype=np.float32)
+        self.cube = set_up_variable_cube(
+            data, name="rainfall_rate", units="mm h-1",
+            spatial_grid="equalarea",
+            time=datetime.datetime(2017, 11, 10, 4, 0),
+            frt=datetime.datetime(2017, 11, 10, 4, 0))
+        self.cube.coord("projection_y_coordinate").points = (
+            self.plugin.y_coord.points)
+        self.cube.coord("projection_y_coordinate").bounds = (
+            self.plugin.y_coord.bounds)
+        self.cube.coord("projection_x_coordinate").points = (
+            self.plugin.x_coord.points)
+        self.cube.coord("projection_x_coordinate").bounds = (
+            self.plugin.x_coord.bounds)
```

Comment:
I'm not sure this is a good place to use the general utility.  For the advect field plugin we will be inputting a cube that is not a forecast, so is likely not to have a forecast reference time or forecast period.  So the simpler cube is probably a better test input here.

Reply:
Reverted to use previous cube set-up.